### ISSUE TITLE
Missing side-effects

### ISSUE THREAD

**Issue by Rich-Harris** [rollup/rollup#39]
Given this situation...

```js
// main.js
import Foo from './Foo';
var foo = new Foo();
var baz = new Foo.Baz();

// Foo.js
import { Bar } from './Bar';
import { Baz } from './Baz';

Bar.Baz = Baz; // this statement is ignored

export default Bar;

// Bar.js
export function Bar () {
  alert( 'bar' );
}

// Baz.js
export function Baz () {
  alert('baz');
}
```

...we'd expect `Foo` to have a `Baz` property, but Rollup goes straight to the original `Foo` definition, ignoring the `Bar.Baz = Baz` line:

```js
function Bar () {
  alert( 'bar' );
}

var Foo = Bar;

var foo = new Foo();
var baz = new Foo.Baz();
```

We need to catch any statements that mutate imported bindings, wherever they are.

**Comment by eventualbuddha** [comment]
Is this fixed now? If so, mind releasing a build with the fix?

**Comment by Rich-Harris** [comment]
Yep, fixed - sorry, got sidetracked, will push out a release momentarily

**Comment by Rich-Harris** [comment]
Released as 0.11.0

**Comment by eventualbuddha** [comment]
Thanks!
